src/org/zaproxy/zap/extension/fuzz/httpfuzzer/processors/tagcreator/HttpFuzzerMessageProcessorTagUIPanel.java

```diff
@@ -0,0 +1,146 @@
+    private static final String MATCH_BY_REGEX_NAME = Constant.messages.getString("fuzz.httpfuzzer.processor.tagcreator.matchbyregex.name");
+    private static final String EXTRACT_BY_REGEX_LABEL_REGEX = Constant.messages.getString("fuzz.httpfuzzer.processor.tagcreator.extractbyregex.regex");
+    private static final String EXTRACT_BY_REGEX_NAME = Constant.messages.getString("fuzz.httpfuzzer.processor.tagcreator.extractbyregex.name");
+    private final GridLayout layout;
+    private final JPanel fieldsPanel;
+    private ButtonGroup radioGroup;
+
+    private JTextField matchByRegExFieldRegEx;
+    private JTextField matchByRegExFieldTag;
+    private JRadioButton matchByRegExRadio;
+
+    private JTextField extractByRegExFieldRegEx;
+    private JRadioButton extractByRegExRadio;
+
+    public HttpFuzzerMessageProcessorTagUIPanel() {
+        fieldsPanel = new JPanel();
+        radioGroup = new ButtonGroup();
+        layout = new GridLayout(5, 2);
+        fieldsPanel.setLayout(layout);
+        createMatchByRegExControl();
+        createExtractByRegExControl();
+    }
+
+    private void createExtractByRegExControl() {
+        extractByRegExRadio = new JRadioButton();
+        addRadioButton(extractByRegExRadio, EXTRACT_BY_REGEX_NAME, false);
+
+        extractByRegExFieldRegEx = new JTextField();
+        addLabelWithTextField(extractByRegExFieldRegEx, EXTRACT_BY_REGEX_LABEL_REGEX);
+    }
+
+    private void createMatchByRegExControl(){
+        matchByRegExRadio = new JRadioButton();
+        addRadioButton(matchByRegExRadio, MATCH_BY_REGEX_NAME, true);
+
+        matchByRegExFieldRegEx = new JTextField();
+        addLabelWithTextField(matchByRegExFieldRegEx, MATCH_BY_REGEX_LABEL_REGEX);
+
+        matchByRegExFieldTag = new JTextField();
+        addLabelWithTextField(matchByRegExFieldTag, MATCH_BY_REGEX_LABEL_TAG);
+    }
+
+    private void addRadioButton(JRadioButton radio, String name, Boolean selected) {
+        radio.setText(name);
+        radio.setSelected(selected);
+        fieldsPanel.add(radio);
+        fieldsPanel.add(new Panel());
+        radioGroup.add(radio);
+    }
+
+    private void addLabelWithTextField(JTextField textField, String labelName){
+        JLabel label = new JLabel(labelName,  JLabel.TRAILING);
+        fieldsPanel.add(label);
+        label.setLabelFor(textField);
+        fieldsPanel.add(textField);
+    }
+
+    @Override
+    public JPanel getComponent() {
+        return fieldsPanel;
+    }
+
+    @Override
+    public void setFuzzerMessageProcessorUI(HttpFuzzerMessageProcessorTagUI processorUI) {
+        ITagRule rule = processorUI.getTagRule();
+        if(rule.getClass() == MatchByRegExTagRule.class){
+            MatchByRegExTagRule tagRule = (MatchByRegExTagRule)rule;
+            matchByRegExRadio.setSelected(true);
+            extractByRegExRadio.setSelected(false);
```

Comment:
This statement is not needed, ButtonGroup already takes care to keep just one button selected at any given time.